### ISSUE TITLE
option to disable "no addresses returned for this domain are permitted"

### ISSUE THREAD

**Issue by northys** [safing/portmaster#98]
**What would you like to be added**:

Option to disable "permitted addresses" to filter DNS responses. I use domain names which resolves to 10.0.0.0/8 IPs on regular basis and I get used to SSH using IP instead of comfortable using an hostname. But the fact that there is no way to access OBB railnet is driving me crazy. When I stop portmaster the DNS on my laptop doesn't work at all and when portmaster is running the `0.0.0.0` is returned.

```
 blocking connection northys:/usr/bin/dig:511278 to railnet.oebb.at. (<nil>), no addresses returned for this domain are permitted
```

**Why is this needed**:

Because we live in 21st century where web is accessed using encrypted HTTPS.

**Comment by northys** [comment]
I've connected using phone tethering and DNS randomly do not resolve some domain while I have full LTE. I'm just curious how can query for google.com result in disallowed address. It looks like my laptop is still resolving using DNS in train or I don't know. but I'm connected only to my Pixel using wi-fi tethering.

I'm curious how switching between DHCP DNS list and my DNS list in portmaster works.

```
sudo journalctl -fu portmaster.service --since "5 minutes ago" | grep -i permitted
čec 16 19:30:54 northys-laptop portmaster-control[513451]: 200716 19:30:54.388 connection:221 ▶ INFO 697 filter: blocking connection northys:/usr/lib/firefox/firefox:512127 to token.services.mozilla.com. (<nil>), no addresses returned for this domain are permitted
čec 16 19:31:06 northys-laptop portmaster-control[513451]: 200716 19:31:06.707 connection:221 ▶ INFO 771 filter: blocking connection northys:/usr/bin/dig:514199 to google.com. (<nil>), no addresses returned for this domain are permitted
čec 16 19:32:11 northys-laptop portmaster-control[513451]: 200716 19:32:11.352 connection:221 ▶ INFO 908 filter: blocking connection northys:/usr/bin/inetutils-traceroute:514339 to sirius.northys.cz. (<nil>), no addresses returned for this domain are permitted
```


Sometimes it works, sometimes it doesn't...
```
sudo journalctl -fu portmaster.service --since "5 minutes ago" | grep -i google.com
čec 16 19:31:01 northys-laptop portmaster-control[513451]: 200716 19:31:01.816 all/master:042 ▶ INFO 707 filter: re-evaluating verdict on northys:/usr/lib/firefox/firefox:512127 to www.google.com. (<nil>)
čec 16 19:31:01 northys-laptop portmaster-control[513451]: 200716 19:31:01.816 connection:207 ▶ INFO 708 filter: granting connection northys:/usr/lib/firefox/firefox:512127 to www.google.com. (<nil>), endpoint is not blacklisted (default=permit)
čec 16 19:31:01 northys-laptop portmaster-control[513451]: 200716 19:31:01.817 all/master:042 ▶ INFO 709 filter: re-evaluating verdict on northys:/usr/lib/firefox/firefox:512127 to www.google.com. (<nil>)
čec 16 19:31:01 northys-laptop portmaster-control[513451]: 200716 19:31:01.817 connection:207 ▶ INFO 710 filter: granting connection northys:/usr/lib/firefox/firefox:512127 to www.google.com. (<nil>), endpoint is not blacklisted (default=permit)
čec 16 19:31:01 northys-laptop portmaster-control[513451]: 200716 19:31:01.817 nameserver:293 ▶ DEBU 711 nameserver: returning response www.google.com.AAAA to northys:/usr/lib/firefox/firefox:512127
čec 16 19:31:01 northys-laptop portmaster-control[513451]: 200716 19:31:01.817 nameserver:293 ▶ DEBU 712 nameserver: returning response www.google.com.A to northys:/usr/lib/firefox/firefox:512127
čec 16 19:31:01 northys-laptop portmaster-control[513451]: 200716 19:31:01.828 connection:207 ▶ INFO 713 filter: granting connection northys:/usr/lib/firefox/firefox:512127 to www.google.com. (216.58.201.100), endpoint is not blacklisted (default=permit)
čec 16 19:31:01 northys-laptop portmaster-control[513451]: 200716 19:31:01.919 connection:207 ▶ INFO 714 filter: granting connection northys:/usr/lib/firefox/firefox:512127 to www.google.com. (216.58.201.100), endpoint is not blacklisted (default=permit)
čec 16 19:31:02 northys-laptop portmaster-control[513451]: 200716 19:31:02.023 connection:207 ▶ INFO 715 filter: granting connection northys:/usr/lib/firefox/firefox:512127 to www.google.com. (216.58.201.100), endpoint is not blacklisted (default=permit)
čec 16 19:31:02 northys-laptop portmaster-control[513451]: 200716 19:31:02.186 all/master:042 ▶ INFO 725 filter: re-evaluating verdict on northys:/usr/lib/firefox/firefox:512127 to www.google.com. (<nil>)
čec 16 19:31:02 northys-laptop portmaster-control[513451]: 200716 19:31:02.186 connection:207 ▶ INFO 726 filter: granting connection northys:/usr/lib/firefox/firefox:512127 to www.google.com. (<nil>), endpoint is not blacklisted (default=permit)
čec 16 19:31:02 northys-laptop portmaster-control[513451]: 200716 19:31:02.186 nameserver:293 ▶ DEBU 727 nameserver: returning response www.google.com.AAAA to northys:/usr/lib/firefox/firefox:512127
čec 16 19:31:02 northys-laptop portmaster-control[513451]: 200716 19:31:02.186 all/master:042 ▶ INFO 728 filter: re-evaluating verdict on northys:/usr/lib/firefox/firefox:512127 to www.google.com. (<nil>)
čec 16 19:31:02 northys-laptop portmaster-control[513451]: 200716 19:31:02.186 connection:207 ▶ INFO 729 filter: granting connection northys:/usr/lib/firefox/firefox:512127 to www.google.com. (<nil>), endpoint is not blacklisted (default=permit)
čec 16 19:31:02 northys-laptop portmaster-control[513451]: 200716 19:31:02.186 nameserver:293 ▶ DEBU 730 nameserver: returning response www.google.com.A to northys:/usr/lib/firefox/firefox:512127
čec 16 19:31:02 northys-laptop portmaster-control[513451]: 200716 19:31:02.207 connection:207 ▶ INFO 731 filter: granting connection northys:/usr/lib/firefox/firefox:512127 to www.google.com. (216.58.201.100), endpoint is not blacklisted (default=permit)
čec 16 19:31:02 northys-laptop portmaster-control[513451]: 200716 19:31:02.449 connection:207 ▶ INFO 732 filter: granting connection northys:/usr/lib/firefox/firefox:512127 to www.google.com. (216.58.201.100), endpoint is not blacklisted (default=permit)
čec 16 19:31:03 northys-laptop portmaster-control[513451]: 200716 19:31:03.480 connection:207 ▶ INFO 754 filter: granting connection Unknown::-1 to www.google.com. (216.58.201.100), endpoint is not blacklisted (default=permit)
čec 16 19:31:03 northys-laptop portmaster-control[513451]: 200716 19:31:03.577 connection:207 ▶ INFO 757 filter: granting connection Unknown::-1 to www.google.com. (216.58.201.100), endpoint is not blacklisted (default=permit)
čec 16 19:31:03 northys-laptop portmaster-control[513451]: 200716 19:31:03.683 connection:207 ▶ INFO 760 filter: granting connection Unknown::-1 to www.google.com. (216.58.201.100), endpoint is not blacklisted (default=permit)
čec 16 19:31:06 northys-laptop portmaster-control[513451]: 200716 19:31:06.707 connection:207 ▶ INFO 770 filter: granting connection northys:/usr/bin/dig:514199 to google.com. (<nil>), endpoint is not blacklisted (default=permit)



# here is the no addresses returned for this domain are permitted log 
čec 16 19:31:06 northys-laptop portmaster-control[513451]: 200716 19:31:06.707 connection:221 ▶ INFO 771 filter: blocking connection northys:/usr/bin/dig:514199 to google.com. (<nil>), no addresses returned for this domain are permitted




čec 16 19:31:36 northys-laptop portmaster-control[513451]: 200716 19:31:36.511 all/master:042 ▶ INFO 839 filter: re-evaluating verdict on northys:/usr/lib/firefox/firefox:512127 to play.google.com. (<nil>)
čec 16 19:31:36 northys-laptop portmaster-control[513451]: 200716 19:31:36.511 connection:207 ▶ INFO 840 filter: granting connection northys:/usr/lib/firefox/firefox:512127 to play.google.com. (<nil>), endpoint is not blacklisted (default=permit)
čec 16 19:31:36 northys-laptop portmaster-control[513451]: 200716 19:31:36.511 nameserver:293 ▶ DEBU 841 nameserver: returning response play.google.com.A to northys:/usr/lib/firefox/firefox:512127
čec 16 19:31:36 northys-laptop portmaster-control[513451]: 200716 19:31:36.776 all/master:042 ▶ INFO 846 filter: re-evaluating verdict on northys:/usr/lib/firefox/firefox:512127 to play.google.com. (<nil>)
čec 16 19:31:36 northys-laptop portmaster-control[513451]: 200716 19:31:36.776 connection:207 ▶ INFO 847 filter: granting connection northys:/usr/lib/firefox/firefox:512127 to play.google.com. (<nil>), endpoint is not blacklisted (default=permit)
čec 16 19:31:36 northys-laptop portmaster-control[513451]: 200716 19:31:36.777 nameserver:293 ▶ DEBU 848 nameserver: returning response play.google.com.A to northys:/usr/lib/firefox/firefox:512127
čec 16 19:32:01 northys-laptop portmaster-control[513451]: 200716 19:32:01.292 connection:207 ▶ INFO 902 filter: granting connection northys:/usr/bin/inetutils-traceroute:514287 to dns.google.com. (8.8.8.8), endpoint is not blacklisted (default=permit)
čec 16 19:32:01 northys-laptop portmaster-control[513451]: 200716 19:32:01.307 all/master:042 ▶ INFO 903 filter: re-evaluating verdict on northys:/usr/bin/inetutils-traceroute:514287 to dns.google.com. (8.8.8.8)
čec 16 19:32:01 northys-laptop portmaster-control[513451]: 200716 19:32:01.307 connection:207 ▶ INFO 904 filter: granting connection northys:/usr/bin/inetutils-traceroute:514287 to dns.google.com. (8.8.8.8), endpoint is not blacklisted (default=permit)
čec 16 19:32:07 northys-laptop portmaster-control[513451]: 200716 19:32:07.361 connection:207 ▶ INFO 905 filter: granting connection northys:/usr/bin/inetutils-traceroute:514287 to dns.google.com. (8.8.8.8), endpoint is not blacklisted (default=permit)
čec 16 19:33:23 northys-laptop portmaster-control[513451]: 200716 19:33:23.587 connection:207 ▶ INFO 939 filter: granting connection northys:/usr/bin/dig:514472 to google.com. (<nil>), endpoint is not blacklisted (default=permit)
čec 16 19:33:23 northys-laptop portmaster-control[513451]: 200716 19:33:23.590 nameserver:293 ▶ DEBU 940 nameserver: returning response google.com.A to northys:/usr/bin/dig:514472
```

I'm using the normal security level.

**Version:**
```
Name: Portmaster
Version: 0.4.12
Commit: tags/v0.4.12-0-gc9ba2a939a07a12d443feda3043081e89df169d2
Build Options: main.go
Build User: user
Build Host: docker
Build Date: 15.07.2020
Build Source: https://github.com/safing/portmaster
```

**Comment by dhaavi** [comment]
Hey @northys, thanks for reporting this. I think everything that happened here is more or less related to a captive portal used by the train.

Question: Did you complete the captive portal procedure on the train?

You maybe were still in the portal when trying to resolve `railnet.oebb.at.`, which then returned a local IP address. This is highly unusual - borderline malicious - behavior. Blocking such responses is regulated by the setting "Filter Out-of-Scope DNS Records", which you can set to only be active at the Extreme level no to be bothered by it in every day use.

There is a related settings called "Filter DNS Records that would be blocked" which also could result in this block message, but I don't think this is the case here.

Please note that both of these settings are currently in Expertise Level Expert and Release Level Beta - so please update you settings accordingly to see them.

While we have improved captive portal support greatly in the last update, there still are a couple more issues that I'll fix soon. If there is a captive portal, you should be able to trigger it by opening `detectportal.firefox.com` in your browser.

I agree with your further conclusion that the bogus (redirecting) dns responses from the train seem to have been cached. We already have some better cache invalidation mechanisms in the `develop` branch, which will be released shortly. Also, we will better handle the caching of such responses in the future.

With the current version, if you open `detectportal.firefox.com` in your browser after you connect to the train's WiFi and complete the portal process, everything should work, as long as you don't have any bogus dns responses in cache already.

**Comment by northys** [comment]
Just notes from train. I'll structure it and follow your post later today.

```
# try to use suggested detectportal before captive portal confirm
northys at northys-laptop in ~
» http detectportal.firefox.com.
HTTP/1.0 302 Moved Temporarily
Connection: close
Content-Length: 0
Date: Fri, 17 Jul 2020 12:34:18 GMT
Location: http://cdwifi.cz?mac=0C:DD:24:78:83:72&longitude=16.597&latitude=48.9975&class=0&url=http%3A%2F%2Fdetectportal.firefox.com%2F
Server: squid/3.1.20
Via: 1.0 IMP-redirect (squid/3.1.20)
X-Cache: MISS from IMP-redirect
X-Cache-Lookup: MISS from IMP-redirect:3129


# ok it goes to cdwifi.cz which actually resolves to 10.0.0.0/8 - why it works in this case? I think OBB has 172.0.0.0/8 or something like this.
northys at northys-laptop in ~
» dig cdwifi.cz

; <<>> DiG 9.16.1-Ubuntu <<>> cdwifi.cz
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 30352
;; flags: qr rd; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0
;; WARNING: recursion requested but not available

;; QUESTION SECTION:
;cdwifi.cz.			IN	A

;; ANSWER SECTION:
cdwifi.cz.		17	IN	A	10.0.0.60

;; Query time: 275 msec
;; SERVER: 127.0.0.53#53(127.0.0.53)
;; WHEN: Pá čec 17 14:34:23 CEST 2020
;; MSG SIZE  rcvd: 52

# dig google before confirming captive portal, works, surprisingly. i expected it to resolve to captive portal page
northys at northys-laptop in ~
» dig google.com

; <<>> DiG 9.16.1-Ubuntu <<>> google.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 9516
;; flags: qr rd; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0
;; WARNING: recursion requested but not available

;; QUESTION SECTION:
;google.com.			IN	A

;; ANSWER SECTION:
google.com.		17	IN	A	172.217.23.238

;; Query time: 167 msec
;; SERVER: 127.0.0.53#53(127.0.0.53)
;; WHEN: Pá čec 17 14:35:28 CEST 2020
;; MSG SIZE  rcvd: 54

# sadly I can't ping the IP
northys at northys-laptop in ~
» ping google.com
PING google.com (172.217.23.238) 56(84) bytes of data.
^C
--- google.com ping statistics ---
3 packets transmitted, 0 received, 100% packet loss, time 2044ms

# confirmed captive portal, now it don't even resolve
northys at northys-laptop in ~
» ping safing.at
^C⏎       

# this resolve was instant, I haven't been on root.cz today yet                                                                                                                                                                                                         northys at northys-laptop in ~
» dig root.cz

; <<>> DiG 9.16.1-Ubuntu <<>> root.cz
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 27391
;; flags: qr rd; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0
;; WARNING: recursion requested but not available

;; QUESTION SECTION:
;root.cz.			IN	A

;; ANSWER SECTION:
root.cz.		17	IN	A	91.213.160.188

;; Query time: 151 msec
;; SERVER: 127.0.0.53#53(127.0.0.53)
;; WHEN: Pá čec 17 14:36:02 CEST 2020
;; MSG SIZE  rcvd: 48

can't ping even with -n. it won't even report a timeout (maybe i didn't wait long enough)
northys at northys-laptop in ~
» ping root.cz
^C⏎                                                                                                                                                                                                                northys at northys-laptop in ~
» ping -n root.cz
PING root.cz (91.213.160.188) 56(84) bytes of data.
^C
--- root.cz ping statistics ---
9 packets transmitted, 0 received, 100% packet loss, time 8197ms

# switched to my phone, internet works
northys at northys-laptop in ~
» ping google.com
PING google.com (172.217.23.238) 56(84) bytes of data.
64 bytes from prg03s06-in-f238.1e100.net (172.217.23.238): icmp_seq=1 ttl=118 time=82.8 ms
64 bytes from prg03s06-in-f238.1e100.net (172.217.23.238): icmp_seq=2 ttl=118 time=62.1 ms
64 bytes from prg03s06-in-f238.1e100.net (172.217.23.238): icmp_seq=3 ttl=118 time=268 ms
64 bytes from prg03s06-in-f238.1e100.net (172.217.23.238): icmp_seq=4 ttl=118 time=111 ms
64 bytes from prg03s06-in-f238.1e100.net (172.217.23.238): icmp_seq=5 ttl=118 time=124 ms
64 bytes from prg03s06-in-f238.1e100.net (172.217.23.238): icmp_seq=6 ttl=118 time=23.3 ms
64 bytes from prg03s06-in-f238.1e100.net (172.217.23.238): icmp_seq=7 ttl=118 time=42.4 ms
64 bytes from prg03s06-in-f238.1e100.net (172.217.23.238): icmp_seq=8 ttl=118 time=54.3 ms
64 bytes from prg03s06-in-f238.1e100.net (172.217.23.238): icmp_seq=9 ttl=118 time=33.8 ms
^C
--- google.com ping statistics ---
9 packets transmitted, 9 received, 0% packet loss, time 8368ms
rtt min/avg/max/mdev = 23.330/89.186/268.376/71.037 ms
```

**Comment by northys** [comment]
I think that the problem is docker again. OBB's trains use the same default subnet as docker - `172.17.0.0/16`. This is the reason why the captive portal won't resolve. I just recalled this fact and I'll try to confirm it asap. Maybe this is the reason why I'm trying to avoid them :D

**Comment by ppacher** [comment]
Hi @northys, have you been able to confirm that the ÖBB network conflicting with docker causes this? As you mentioned working with kubernetes I guess you know how to get your head around docker. Just in case, you can change the docker network to something less common by editing/creating `/etc/docker/daemon.json` and putting the following in it (change the IP address as you like):
```json
{
    "bip": "10.211.0.1/24"
}
```
You finally need to restart the docker daemon, on a systemd machine the following should do the trick: `systemctl restart docker`. 

> Note that this will cause networking issues for already running containers!

**Comment by northys** [comment]
@ppacher I gonna travel with OBB next friday. I've made an appointment in my calendar to don't forget about this ticket.

I've changed my docker subnet as you suggested, thanks.

**Comment by northys** [comment]
@ppacher finally i got to test it. it works and the problem was conflicting docker subnet as mentioned earlier.

**Comment by northys** [comment]
1) Is there any sort of documentation? I would love it if this one could be part of it. I've spent ~10m now figuring out where we solved this and it also took me some time before I found out how to make `Filter DNS Records that would be blocked` toggle visible.
2) I've enabled this and digged multiple times with strange results. Each dig returned different response until it finally returned the correct one.
3) We need to figure out how can I send you raw logs directly because I'm not going to clean every log I'm sending you. My greps make it out of context, and you always see nothing in it after I clean it. Also, I would prefer nobody here knowing what domains my computer queries in background =) 

```
northys at northys-laptop in /var
» dig 8.8.8.8

; <<>> DiG 9.16.1-Ubuntu <<>> 8.8.8.8
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: REFUSED, id: 31952
;; flags: qr rd; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 0
;; WARNING: recursion requested but not available

;; QUESTION SECTION:
;8.8.8.8.			IN	A

;; Query time: 1484 msec
;; SERVER: 127.0.0.53#53(127.0.0.53)
;; WHEN: Ne zář 13 21:56:27 CEST 2020
;; MSG SIZE  rcvd: 25

northys at northys-laptop in /var
» dig nuc.vpn.northys.cz

; <<>> DiG 9.16.1-Ubuntu <<>> nuc.vpn.northys.cz
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 18781
;; flags: qr rd; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0
;; WARNING: recursion requested but not available

;; QUESTION SECTION:
;nuc.vpn.northys.cz.		IN	A

;; ANSWER SECTION:
nuc.vpn.northys.cz.	0	IN	A	0.0.0.0

;; Query time: 212 msec
;; SERVER: 127.0.0.53#53(127.0.0.53)
;; WHEN: Ne zář 13 22:00:57 CEST 2020
;; MSG SIZE  rcvd: 70

northys at northys-laptop in /var
» dig nuc.vpn.northys.cz

; <<>> DiG 9.16.1-Ubuntu <<>> nuc.vpn.northys.cz
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 20478
;; flags: qr rd; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 0
;; WARNING: recursion requested but not available

;; QUESTION SECTION:
;nuc.vpn.northys.cz.		IN	A

;; Query time: 204 msec
;; SERVER: 127.0.0.53#53(127.0.0.53)
;; WHEN: Ne zář 13 22:00:58 CEST 2020
;; MSG SIZE  rcvd: 36

northys at northys-laptop in /var
» dig nuc.vpn.northys.cz

; <<>> DiG 9.16.1-Ubuntu <<>> nuc.vpn.northys.cz
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 50907
;; flags: qr rd; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 0
;; WARNING: recursion requested but not available

;; QUESTION SECTION:
;nuc.vpn.northys.cz.		IN	A

;; Query time: 204 msec
;; SERVER: 127.0.0.53#53(127.0.0.53)
;; WHEN: Ne zář 13 22:00:59 CEST 2020
;; MSG SIZE  rcvd: 36

northys at northys-laptop in /var
» dig nuc.vpn.northys.cz

; <<>> DiG 9.16.1-Ubuntu <<>> nuc.vpn.northys.cz
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 62456
;; flags: qr rd; QUERY: 1, ANSWER: 1, AUTHORITY: 5, ADDITIONAL: 17
;; WARNING: recursion requested but not available

;; QUESTION SECTION:
;nuc.vpn.northys.cz.		IN	A

;; ANSWER SECTION:
nuc.vpn.northys.cz.	17	IN	A	10.10.0.3

;; AUTHORITY SECTION:
northys.cz.		17	IN	NS	ns3.gransy.com.
northys.cz.		17	IN	NS	ns4.gransy.com.
northys.cz.		17	IN	NS	ns5.gransy.com.
northys.cz.		17	IN	NS	ns2.gransy.com.
northys.cz.		17	IN	NS	ns.gransy.com.

;; ADDITIONAL SECTION:
ns.gransy.com.		17	IN	A	77.78.104.149
ns.gransy.com.		17	IN	A	95.179.136.180
ns.gransy.com.		17	IN	A	45.76.90.43
ns2.gransy.com.		17	IN	A	217.69.4.49
ns2.gransy.com.		17	IN	A	95.179.192.121
ns2.gransy.com.		17	IN	A	217.16.185.221
ns3.gransy.com.		17	IN	A	77.78.126.35
ns3.gransy.com.		17	IN	A	109.74.7.71
ns3.gransy.com.		17	IN	A	46.21.100.103
ns4.gransy.com.		17	IN	A	66.42.94.136
ns4.gransy.com.		17	IN	A	80.67.7.209
ns5.gransy.com.		17	IN	A	31.192.227.223
ns.gransy.com.		17	IN	AAAA	2001:19f0:5001:1caf:5400:1ff:fec5:8d50
ns2.gransy.com.		17	IN	AAAA	2001:19f0:6801:474:5400:1ff:fec5:8efb
ns2.gransy.com.		17	IN	AAAA	2001:19f0:7401:81e0:5400:1ff:fec5:8ef6
ns3.gransy.com.		17	IN	AAAA	2a02:750:7::2e7
ns3.gransy.com.		17	IN	AAAA	2a02:750:8::32

;; Query time: 452 msec
;; SERVER: 127.0.0.53#53(127.0.0.53)
;; WHEN: Ne zář 13 22:01:00 CEST 2020
;; MSG SIZE  rcvd: 825
```

**Comment by dhaavi** [comment]
> 1. Is there any sort of documentation? I would love it if this one could be part of it. I've spent ~10m now figuring out where we solved this and it also took me some time before I found out how to make `Filter DNS Records that would be blocked` toggle visible. EDIT: its actually the " Filter Out-of-Scope DNS Records" toggle

I think we need more comments in the extra section, as just mentioned in the other issue. 😁

> 2. I've enabled this and digged multiple times with strange results. Each dig returned different response until it finally returned the correct one.

We found a 🐞! What seemed to happen here is that a filtered response probably leaked into the cache. You then resetted the cache by querying three times in a row (I love that feature!).

Reference to new issue: https://github.com/safing/portmaster/issues/144

> 3. We need to figure out how can I send you raw logs directly because I'm not going to clean every log I'm sending you. My greps make it out of context, and you always see nothing in it after I clean it. Also, I would prefer nobody here knowing what domains my computer queries in background =)

I will bring this up with the team and see if we can find a better solution for this - there must be!

**Comment by northys** [comment]
@dhaavi is the option `Filter Out-of-Scope DNS Records` missing after redesign? I can't find it anymore but I would love to enable domains to be resolved into internal IP (e.g. 10.0.0.0/8).

**Comment by dhaavi** [comment]
It was renamed to `Enforce Global/Private Split-View`. The cog icon next to a blocked connection / DNS request should take you right to it, if you are looking at it in the UI.

**Comment by dhaavi** [comment]
For future readers: The [Enforce Global/Private Split-View](https://docs.safing.io/portmaster/settings#filter/removeOutOfScopeDNS) settings has been moved after the [Outgoing Rules](https://docs.safing.io/portmaster/settings#filter/endpoints) setting, so you can now just add a domain to the rules to circumvent it being blocked by the split view enforcement.

I'm also closing this as this seems to be fully resolved now.
